frontend/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -137,6 +137,19 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
   static getDeckObject = (props: PropsWithHeight): DeckObject => {
     const { element, width, height } = props
     const json = JSON.parse(element.json)
+    let theme
+    let mapTheme
+
+    // Use either the Mapbox light or dark style based on Streamlit's theme
+    // For Mapbox styles, see https://docs.mapbox.com/api/maps/styles/#mapbox-styles
+    if ("stActiveTheme-/-v1" in localStorage) {
```

Comment:
Instead of getting the theme info from the local storage, an alternative (an maybe better) way would be to use the `withTheme`  hook to inject the theme into the component and check the background of the configured theme for being dark or light. `ArrowVegaLiteChart` is doing this:

https://github.com/streamlit/streamlit/blob/786da40420cd5b6d7682da42837c6ecf861e8464/frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx#L507

https://github.com/streamlit/streamlit/blob/786da40420cd5b6d7682da42837c6ecf861e8464/frontend/src/components/elements/ArrowVegaLiteChart/CustomTheme.tsx#L24

Reply:
Due to my limited JS knowledge I wasn't sure what those snippets exactly did. I'll try incorporating your suggestions in upcoming commits ❤️

Reply:
Unfortunately, it's still a class-based components which makes it a bit more complicated. I will add a few more lines that you might need to add as well:

2) Import `withTheme` and `Theme`:
https://github.com/streamlit/streamlit/blob/786da40420cd5b6d7682da42837c6ecf861e8464/frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx#L19
1) add `theme` to props:

https://github.com/streamlit/streamlit/blob/786da40420cd5b6d7682da42837c6ecf861e8464/frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx#L55

Reply:
Should be fixed on 3506f07